libraries/AC_PrecLand/AC_PrecLand_Companion.cpp

```diff
@@ -46,9 +46,19 @@ void AC_PrecLand_Companion::handle_msg(const mavlink_landing_target_t &packet, u
 {
     _distance_to_target = packet.distance;
 
-    // compute unit vector towards target
-    _los_meas_body = Vector3f(-tanf(packet.angle_y), tanf(packet.angle_x), 1.0f);
-    _los_meas_body /= _los_meas_body.length();
+    if (packet.position_valid == 1) {
+        if (packet.frame == MAV_FRAME_BODY_FRD) {
```

Comment:
need to clarify, this usage seems to assume this is body frame, but MAV_FRAME_BODY_FRD in common.xml seems to imply this as a frame whose roll/pitch components are level with earth

Reply:
Hi @tridge Thank you for reviewing. I agree with you, but I am afraid that MAV_FRAME_BODY_FRD may be the closest choice.

Reply:
Hi @tridge I think we use MAV_FRAME_BODY_OFFSET_NED for handling [SET_POSITION_TARGET_LOCAL_NED](https://ardupilot.org/dev/docs/copter-commands-in-guided-mode.html#set-position-target-local-ned) in guided mode. But MAV_FRAME_BODY_OFFSET_NED  is replaced by MAV_FRAME_BODY_FRD in current mavlink doc.

Reply:
> need to clarify, this usage seems to assume this is body frame, but MAV_FRAME_BODY_FRD in common.xml seems to imply this as a frame whose roll/pitch components are level with earth

IMHO MAV_FRAME_BODY_FRD related to body frame including rill and pitch. But MAV_FRAME_LOCAL_FRD with z axis to earh (so it with level with earth). I didnt find 100% clear description related to it.